core/src/main/java/com/g16/feyrune/model/player/Player.java

```diff
@@ -53,8 +54,8 @@ public Texture getTexture() {
     }
 
     //TODO: This is wrong and temp
-    public ICreature getCreature(){
-        return creature;
+    public PlayerCreature getCreature(){
+        return new PlayerCreature((BaseCreature) this.creature);
```

Comment:
Confusing function name... Either rename to getPlayerCreature or change to return a creature

Reply:
Agreed